src/Mod/Import/App/dxf/dxf.cpp

```diff
@@ -3271,7 +3280,118 @@ bool CDxfRead::ReadLayer()
     return false;
 }
 
-void CDxfRead::DoRead(const bool ignore_errors /* = false */ )
+bool CDxfRead::ReadVersion()
+{
+    static const std::vector<std::string> VersionNames = {
+        // This table is indexed by eDXFVersion_t - (ROlder+1)
+        "AC1006",
+        "AC1009",
+        "AC1012",
+        "AC1014",
+        "AC1015",
+        "AC1018",
+        "AC1021",
+        "AC1024",
+        "AC1027",
+        "AC1032"};
+
+    assert(VersionNames.size() == RNewer - ROlder - 1);
+    get_line();
+    get_line();
+    std::vector<std::string>::const_iterator first = VersionNames.cbegin();
+    std::vector<std::string>::const_iterator last = VersionNames.cend();
+    std::vector<std::string>::const_iterator found = std::lower_bound(first, last, m_str);
+    if (found == last)
+        m_version = RNewer;
+    else if (*found == m_str)
+        m_version = (eDXFVersion_t)(std::distance(first, found) + (ROlder + 1));
+    else if (found == first)
+        m_version = ROlder;
+    else
+        m_version = RUnknown;
+
+    return ResolveEncoding();
+}
+
+bool CDxfRead::ReadDWGCodePage()
+{
+    get_line();
+    get_line();
+    assert(m_CodePage == NULL); // If not, we have found two DWGCODEPAGE variables or DoRead was called twice on the same CDxfRead object.
+    m_CodePage = new std::string(m_str);
+
+    return ResolveEncoding();
+}
+
+bool CDxfRead::ResolveEncoding()
+{
+    if (m_encoding != NULL) {
+        delete m_encoding;
+        m_encoding = NULL;
+    }
+    if (m_version >= R2007) {   // Note this does not include RUnknown, but does include RLater
+        m_encoding = new std::string("utf_8");
+        stringToUTF8 = &CDxfRead::UTF8ToUTF8;
+    }
+    else if (m_CodePage == NULL) {
+        // cp1252
+        m_encoding = new std::string("cp1252");
+        stringToUTF8 = &CDxfRead::GeneralToUTF8;
+    }
+    else {
+        // Names may be of the form "ansi_1252" which we map to "cp1252" but we don't map "ansi_x3xxxx" which means "ascii"
+        if (strncmp(m_CodePage->c_str(), "ansi_", 5) == 0
+            && strncmp(m_CodePage->c_str(), "ansi_x3", 7) != 0) {
+            std::string* p = new std::string(*m_CodePage);
+            p->replace(0, 5, "cp");
+            m_encoding = p;
+        }
+        else
```

Comment:
The `if` has curly braces, so add it to the `else` branch too, please.

Reply:
If you insist I will, but I personally abhor unnecessary braces (in particular, the extra lines which they occupy, meaning less code fits on a screen)

Reply:
Actually it is moot now as I've recoded it to not need braces at all.

However, for future reference, is there a document describing style guidelines?

Does that apply in the opposite, i.e. does
```
if (condition)
    simple statement
else {
    many statements
    on several lines
}
```
need braces around the `simple statement`?

What about a chain of `else if`s, does one only have to use braces after the first compound statement?

What about if the `simple statement` is so long that it is wrapped onto multiple lines?

Reply:
In many code style guidelines there is a rule that if a branch of an if-else block requires curly braces then it should be added to all of them.